### ISSUE TITLE
Regression: Reanalyze with (dynamic) variable/constant BPM does no longer honor override

### ISSUE THREAD

**Issue by fwcd** [mixxxdj/mixxx#11872]
### Bug Description

Right-clicking a track and selecting `Reanalyze > Reanalyze (variable BPM)` (or `Reanalyze (constant BPM)` ) does not honor this override and uses the user setting instead.

This regression seems to have been introduced in https://github.com/mixxxdj/mixxx/compare/a547664420463515926a619aafe08b378b34e035...f58dede75953654740a0d66c62af556e6bffd85e:

```
Last good: a547664420 from 2023-08-13 21:20:02 +0200 (Merge branch '2.4' of github.com:mixxxdj/mixxx)
First bad: f58dede759 from 2023-08-17 23:59:45 +0200 (Merge branch '2.4' of github.com:mixxxdj/mixxx)
```

### Version

f58dede759

### OS

macOS 13.5

**Comment by fwcd** [comment]
This seems to have introduced the issue:

https://github.com/mixxxdj/mixxx/commit/1d7eadb529ce6ec574b5ad805a1282bcddbbeaa9#diff-d5a2056b2bfeb0495c1927aaf64f90ff33b1557f4fc13ad42a03ba9337210e9dL146-R147

Since `AnalyzerTrack` can implicitly be constructed from `TrackPointer` this is easy to overlook. (Perhaps we should make the `AnalyzerTrack` constructor `explicit`?)

@Holzhaus

**Comment by Swiftb0y** [comment]
Thank you. closed by #11873 IIUC
